File: 3217. Delete Nodes From Linked List Present in Array/3217. Delete Nodes From Linked List Present in Array.cpp
class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) {
        unordered_set<int> hashSet;

        for(int& num : nums)
            hashSet.insert(num);

        for(ListNode *ptr = head; ptr->next;){
            if(hashSet.find(ptr->next->val) != hashSet.end())
                ptr->next = ptr->next->next;
            else ptr = ptr->next;
        }

        if(hashSet.find(head->val) != hashSet.end())
            head = head->next;

        return head;
    }
};

File: 3217. Delete Nodes From Linked List Present in Array/3217. Delete Nodes From Linked List Present in Array.py
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        hashSet = set()

        for num in nums:
            hashSet.add(num)

        ptr = head
        while ptr.next:
            if ptr.next.val in hashSet:
                ptr.next = ptr.next.next
            else: ptr = ptr.next

        if head.val in hashSet:
            head = head.next

        return head